includes/class-order.php

```diff
@@ -33,6 +33,70 @@ class Order {
 	public static $requiring_states = array( 'AU', 'CA', 'US' );
 
 
+	/**
+	 * Register additional order statuses.
+	 *
+	 * @param  array $order_statuses WC Order statuses.
+	 * @return array
+	 */
+	public static function register_shop_order_post_statuses( $order_statuses ) {
+		$zota_order_statuses = array(
+			'wc-partial-payment' => array(
+				'label'                     => _x( 'Partial Payment', 'Order status', 'zota-woocommerce' ),
+				'public'                    => false,
+				'exclude_from_search'       => false,
+				'show_in_admin_all_list'    => true,
+				'show_in_admin_status_list' => true,
+				/* translators: %s: number of orders */
+				'label_count'               => _n_noop( 'Partial Payment <span class="count">(%s)</span>', 'Partial Payment <span class="count">(%s)</span>', 'zota-woocommerce' ),
+			),
+			'wc-overpayment'    => array(
+				'label'                     => _x( 'Overpayment', 'Order status', 'zota-woocommerce' ),
+				'public'                    => false,
+				'exclude_from_search'       => false,
+				'show_in_admin_all_list'    => true,
+				'show_in_admin_status_list' => true,
+				/* translators: %s: number of orders */
+				'label_count'               => _n_noop( 'Overpayment <span class="count">(%s)</span>', 'Overpayment <span class="count">(%s)</span>', 'zota-woocommerce' ),
+			),
+		);
+
+		return apply_filters( 'wc_gateway_zota_register_shop_order_post_statuses', array_merge( $order_statuses, $zota_order_statuses ) );
+	}
+
+
+	/**
+	 * Register additional order statuses.
+	 *
+	 * @param  array $order_statuses WC Order statuses.
+	 * @return array
+	 */
+	public static function valid_order_statuses_for_payment_complete( $order_statuses ) {
+		$zota_order_statuses = array(
+			'partial-payment',
+			'overpayment',
+		);
```

Comment:
The point is little different. We need the status to complete the payment. The status `partial-payment` needs to be valid for changing from `partial-payment` to `processing` or `completed`. Here is the reason:
https://github.com/woocommerce/woocommerce/blob/9834ecec75573c41530efdbf241b49f9afba4060/includes/class-wc-order.php#L111
Maybe the opposite question is more suitable: Do we need to include 'overpayment' in valid statuses for order payment completion? Because it can't go from overpayment to original amount...

Reply:
Dearest @Lera-S,

can you please take a look at @atanasantonov's question on overpayment and order status?

Reply:
@atanasantonov yes, "overpayment" should be among woocommerce_valid_order_statuses_for_payment_complete, cuz order was fully paid, so platform admin should be able to finalize the order. As you've mentioned, it can't go from overpayment to normal.

Reply:
Ok @Lera-S make sense. Suppose that is ok as it is.